src/components/database/Queries.js

```diff
@@ -0,0 +1,274 @@
+        .where(field, '==', value)
+        .get();
+
+    return data.docs
+        .map((doc: any): any => doc.data())
+        .map((doc: any): UserType | PendingType | ActionType =>
+          sanitize(collection, doc));
+  } catch (error) {
+    console.log(error);
+    return [];
+  }
+}
+
+/**
+  * deleteDocument deletes a document in a specified collection
+  * @param {string} collection defines the collection
+  * @param {'email' | 'count'} field specifies the search criteria for the doc
+  * @param {string | number} value specifies the value of the field
+ */
+async function deleteDocument(collection: string, field: 'email' | 'count',
+    value: string | number) {
+  try {
+    await database.runTransaction(async (t: any) => {
+      const collectionRef = database.collection(collection);
+      const snapshot = await collectionRef
+          .where(field, '==', value)
+          .get();
+      const refs = snapshot.docs.map((doc: any): any => doc.ref);
+
+      await t.delete(refs[0]);
+    });
+  } catch (error) {
+    console.log(error);
+    throw new Error('Delete unsuccesful.');
+  }
+}
+
+/**
+  * updateAdminNote allows admins to add notes to active users
+  * @param {'email' | 'count'} field specifies the field to search for
+  * @param {string | number} value specifies the value of the field
+  * @param {string} newNote specifies the note to be added to the user
+ */
+async function updateAdminNote(field: 'email' | 'count',
+    value: string | number, newNote: string) {
+  try {
+    await database.runTransaction(async (t: any) => {
+      const membersRef = database.collection('active-members');
+      const snapshot = await membersRef
+          .where(field, '==', value)
+          .get();
+
+      const refs = snapshot.docs.map((doc: any): any => doc.ref);
+      const userData = snapshot.docs
+          .map((doc: any): any => doc.data())
+          .map((doc: any): UserType | PendingType | ActionType =>
+            sanitize('active-members', doc));
+
+      await t.update(refs[0], {'adminNote': newNote});
+
+      const counterRef = database.collection('counters').doc('actions');
+      const {counter} = (await counterRef.get()).data();
+      const actionRef = database.collection('actions');
+      const newActionRef = actionRef.doc();
+      await t.set(newActionRef,
+          actionObject(counter + 1, userData[0], 'NEEDS ATTENTION'));
+      await t.update(counterRef, {'counter': counter + 1});
+    });
+  } catch (error) {
+    console.log('Update unsuccesful');
+    throw new Error(error);
+  }
+}
+
+/**
+  * movePendingUser confirms membership of a user and
+  * moves a user from the pending collection to the active collection
+  * @param {'email' | 'count'} field by which the user is looked up
+  * @param {string | number} value of the search field
+ */
+async function movePendingUser(field: 'email' | 'count',
+    value: string | number) {
+  try {
+    await database.runTransaction(async (t: any) => {
```

Comment:
Since it's "any" type, could you please make this name more meaningful?
We are doing many operations on it in later parts of code.
https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers